### PR TITLE
TOMEE-2087 IvmContext.list() does not correctly list the context content

### PR DESCRIPTION
**Pull request by SvetlinZarev**
* MyNamingEnumeration.gatherNodes() adds the wrong federated context entries in the result set (related to TOMEE-2086 - it was indeed a typo, but we "fixed" it with the wrong NameNode).

* MyNamingEnumeration.isMyChild() considers entries that are NOT children to the "parent" tree as such

* Wrong parentNode is passed as argument to gatherNodes in case we are listing the context for any "IvmContext != this

### CODE REVIEW

File: arquillian/arquillian-tomee-tests/arquillian-tomee-webprofile-tests/src/test/java/org/apache/openejb/arquillian/tests/naming/IvmContextTest.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+@RunWith(Arquillian.class)
+public class IvmContextTest {
+    private static final Logger logger = Logger.getLogger(IvmContextTest.class.getName());
+    private static final String TEST_NAME = IvmContextTest.class.getSimpleName();
+    private static final String RESOURCE_EJB_JAR_XML = "ejb-jar.xml";
+    private static final String CONTENT_LOCATION_EJB_JAR_XML = "org/apache/openejb/arquillian/tests/naming/list-context-ejbjar.xml";
+
+    @EJB
+    NamingBean namingBean;
```
Comment (otaviojava) [inline arquillian/arquillian-tomee-tests/arquillian-tomee-webprofile-tests/src/test/java/org/apache/openejb/arquillian/tests/naming/IvmContextTest.java]: Is there any problem, if namingBean be a private attribute?

File: arquillian/arquillian-tomee-tests/arquillian-tomee-webprofile-tests/src/test/java/org/apache/openejb/arquillian/tests/naming/NamingBean.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        logSink.flush();
+
+        if (hasErrors) {
+            throw new IllegalStateException("Failed to lookup some of the listed entries:\n" + getPrintedJndiTree(buffer));
+        }
+    }
+
+    private String getPrintedJndiTree(ByteArrayOutputStream buffer) {
+        try {
+            return buffer.toString("utf-8");
```
Comment (otaviojava) [inline arquillian/arquillian-tomee-tests/arquillian-tomee-webprofile-tests/src/test/java/org/apache/openejb/arquillian/tests/naming/NamingBean.java]: Maybe a constant.
:warning: `buffer.toString(StandardCharsets.UTF_8.name());`

https://docs.oracle.com/javase/7/docs/api/java/nio/charset/StandardCharsets.html#UTF_8

Comment (SvetlinZarev) [inline arquillian/arquillian-tomee-tests/arquillian-tomee-webprofile-tests/src/test/java/org/apache/openejb/arquillian/tests/naming/NamingBean.java]: I'll change it to use the constant

File: arquillian/arquillian-tomee-tests/arquillian-tomee-webprofile-tests/src/test/java/org/apache/openejb/arquillian/tests/naming/IvmContextTest.java (hunk 1 above)
Comment (SvetlinZarev) [inline arquillian/arquillian-tomee-tests/arquillian-tomee-webprofile-tests/src/test/java/org/apache/openejb/arquillian/tests/naming/IvmContextTest.java]: No, I will make it private

**Comment by jeanouii** [comment]
Hi,

As discussed and agreed on the mailing list, I'll be merging this PR on master.
Thanks you very much for you contribution, code wise, but also on the mailing list.

JLouis

**Pull request merged**